### DISCUSSION TITLE
How to avoid transitions between regions in parallel states?

### DISCUSSION THREAD

**Discussion by andywpt** [statelyai/xstate#4514] [Q&A] [answered]
In this parallel states [documentation](https://stately.ai/docs/parallel-states#modeling) , it says "avoid transitions between regions" without giving more elaboration.
Let's say an order has three regions: `paymentStatus`, `orderStatus`, and `deliveryStatus`, and the goal is 

> Whenever `deliveryStatus` changed to **delivered**, change the state of `orderStatus` to **completed**

One approach is to use the **raise** method inside `deliveryStatus`, something like the following:
```
const orderMachine = createMachine({
  id: "order",
  type: "parallel",
  states: {
    orderStatus: {
      initial: "created",
      states: {
        created: {
          on: {
            PACKAGE_DELIVERED: "completed"
          }
        },
        completed: {}
      }
    },
    deliveryStatus: {
      initial: "shipped",
      states: {
        shipped: {
          on: {
            DELIVERED: "delivered"
          },
        },
        delivered: {
          entry: raise("PACKAGE_DELIVERED") 
        }
      }
    }
  },
});
```

But I'm not sure if that violates the "avoid transitions between regions" rule.
Is there a better approach? Any advise would be helpful.

**Comment by Andarist** [comment, marked as answer]
> avoid transitions between regions

I think this is more about how you can prevent certain transitions from happening when a different region is in a specific state or about how you might want to avoid `target: 'someStateInADifferentRegion'`

> One approach is to use the raise method inside deliveryStatus, something like the following:

This is perfect 👌
